test/clojure/core_test/boolean.cljc

```diff
@@ -0,0 +1,56 @@
+(ns clojure.core-test.boolean
+  (:require [clojure.test :as t :refer [deftest testing is are]]
+            [clojure.core-test.number-range :as r]))
+
+(deftest test-false?
+  (are [expected x] (= expected (boolean x))
+    true  0
+    true  1
+    true  -1
+    true  r/max-int
+    true  r/min-int
+    true  0.0
+    true  1.0
+    true  -1.0
+    true  (float 0.0)
+    true  (float 1.0)
+    true  (float -1.0)
+    true  (double 0.0)
+    true  (double 1.0)
+    true  (double -1.0)
+    true  r/max-double
+    true  r/min-double
+    true  ##Inf
+    true  ##-Inf
+    true  ##NaN
+    true  0N
+    true  1N
+    true  -1N
+    true  0/2
+    true  1/2
+    true  -1/2
+    true  0.0M
+    true  1.0M
+    true  -1.0M
+    false nil
+    true  true
+    false false
+    true  "a string"
+    true  "0"
+    true  "1"
+    true  "-1"
+    true  "true"
+    true  "false"
+    true  {:a :map}
+    true  #{:a-set}
+    true  [:a :vector]
+    true  '(:a :list)
+    true  \0
+    true  \1
+    true  :a-keyword
+    true  :true
+    true  :false
+    true  :0
+    true  :1
+    true  :-1
```

Comment:
do you need 6 keywords? this seems to be testing a dialect's reader, not the function implementation, which should be covered by other tests.

Reply:
same with the rest of the tests

Reply:
Not strictly. But cases like those are where weird things often come up. What I'm looking for is whether something is silently converted to something else in the platform, for whatever reason. Think of all the issues that JavaScript has had over the years with odd conversions. I'd note that `:0` and the other numeric ones aren't even strictly legal in Clojure (symbols and keywords are supposed to start with a non-numeric). But Clojure allows them, so it would be interesting to see what other implementations do. But sure, if you feel strongly about it, they aren't strictly necessary and can be removed.